### PR TITLE
Add motor command to sonrd with login/register subcommands

### PR DESCRIPTION
**Pull request by prdlk**
[See demo](https://www.loom.com/share/d313db7b1a1f45a3a88cbac221110792). Initial setup of motor command to create/login a Sonr account pulled from speedway and enhanced with Keychain, table logging.

### New Features *(required)*

* UserAuth Info gets stored in system keychain on `sonrd motor register`
* User is prompted with System Dialog on the first login to access Vault Password, AESDSC Key, and AESPSK Key
* Support for multiple accounts on system

## Checklist

* [ X ] Unit tests
* [ X ] Documentation

Fixes
Connects

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

File: cmd/sonrd/main.go [hunk 1]
```diff
@@ -43,8 +40,7 @@ func main() {
 		app.New,
 		// this line is used by starport scaffolding # root/arguments
 	)
-	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.cli.yaml)")
```
Comment (ntindle) [inline cmd/sonrd/main.go:46]: Why removed

File: cmd/sonrd/utils/keychain.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	"encoding/gob"
+
+	"github.com/pkg/errors"
+	"github.com/sonr-io/keyring"
+	"github.com/sonr-io/sonr/pkg/crypto/mpc"
+	"github.com/sonr-io/sonr/pkg/fs"
+	mt "github.com/sonr-io/sonr/third_party/types/motor"
+)
+
+const K_SERVICE_NAME = "sonr-dev"
```
Comment (ntindle) [inline cmd/sonrd/utils/keychain.go:14]: ?

File: cmd/sonrd/utils/prompt.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	t.AppendRows([]table.Row{
+		{m.GetAddress(), m.GetDID().String(), m.GetBalance()},
+	})
+	t.SetStyle(table.StyleLight)
+	t.Render()
+	golog.Default.Printf("✅ SUCCESS: %s", msg)
+}
+
+func PromptPassword() (string, error) {
+	validate := func(input string) error {
```
Comment (ntindle) [inline cmd/sonrd/utils/prompt.go:43]: Pass this in as an external func

**Review by ntindle — CHANGES_REQUESTED** [review]

File: cmd/sonrd/main.go (hunk 1 above)
Comment (prdlk) [reply cmd/sonrd/main.go:46]: Wasnt me was like this before - https://github.com/sonr-io/sonr/blob/04960e15fed064a7f6ebdf2c7ab82242dc6cf0fa/cmd/sonrd/main.go#L46

File: cmd/sonrd/utils/keychain.go (hunk 2 above)
Comment (prdlk) [reply cmd/sonrd/utils/keychain.go:14]: Keychain service name - will be easier to query across motor nodes

File: cmd/sonrd/utils/prompt.go (hunk 3 above)
Comment (prdlk) [reply cmd/sonrd/utils/prompt.go:43]: can you elaborate?

**Review by ntindle — APPROVED** [review]

**Comment by joshLong145** [comment]
added @SCKelemen

File: cmd/sonrd/utils/info.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	if err != nil {
+		return uuid.NewString()
+	} else {
+		return mid
+	}
+}
+
+// Arch returns the current architecture.
+func Arch() string {
+	return runtime.GOARCH
```
Comment (SCKelemen) [inline cmd/sonrd/utils/info.go:21]: Is this function really necessary?

File: cmd/sonrd/utils/keychain.go [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+type UserAuth struct {
```
Comment (SCKelemen) [inline cmd/sonrd/utils/keychain.go:55]: Is this the struct representing the authorization form, or are we also storing this struct?

File: cmd/sonrd/utils/keychain.go [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+func (i UserAuth) Validate() bool {
+	if len(i.Password) < 12 {
+		return false
+	}
+	if i.AesDSCKey == nil {
+		return false
+	}
+	if len(i.AesDSCKey) < 32 {
```
Comment (SCKelemen) [inline cmd/sonrd/utils/keychain.go:68]: This is weird.

AES Keys can only be 128, 192, or 256 bits, meaning this key has to be 32 bytes or 256 bits. Therefore, should this check not be strict equality?

File: cmd/sonrd/utils/keychain.go [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+		return false
+	}
+	return true
+}
+
+func (i UserAuth) GenAccountCreateRequest() (*mt.CreateAccountRequest, error) {
+	if i.Validate() {
+		return &mt.CreateAccountRequest{
+			Password:  i.Password,
```
Comment (SCKelemen) [inline cmd/sonrd/utils/keychain.go:77]: Assuming here this is the password in flight. Please let me know if this assumption is incorrect.

Comment (ntindle) [reply cmd/sonrd/utils/keychain.go:77]: Can you clarify in flight?

File: cmd/sonrd/utils/keychain.go (hunk 6 above)
Comment (prdlk) [reply cmd/sonrd/utils/keychain.go:68]: @mcjcloud ?

File: cmd/sonrd/utils/keychain.go (hunk 7 above)
Comment (SCKelemen) [reply cmd/sonrd/utils/keychain.go:77]: sent across the wire for authentication purposes, ie a web login form

**Comment by ntindle** [comment]
Are we still planning on doing this?

**Comment by prdlk** [comment]
> Are we still planning on doing this?

Same method with same structure should be in speedway. @ntindle This doesnt use binding layer.

**Comment by mcjcloud** [comment]
Closing this PR for now for inactivity.

**Pull request closed without merging**